### ISSUE TITLE
how to do data augumentation?

### ISSUE THREAD

**Issue by Joanne513** [matterport/Mask_RCNN#2248]

**Comment by konstantin-frolov** [comment]
in the "augmentation" field in the train method, you can transfer an object from the imgaug library. MRCNN supports rotation (affine) and horizontal and vertical reflections (Fliplr, Flipud)

**Comment by yustiks** [comment]
> rotation

Thank you, Konstantin!
Is it any examples available?

**Comment by konstantin-frolov** [comment]
You can create augmentation object and pass it to the "augmentation" field of model.train
```
augmentation = imgaug.augmenters.Sometimes(0.5, imgaug.augmenters.OneOf([imgaug.augmenters.Fliplr(0.5),
                                                                         imgaug.augmenters.Flipud(0.5),
                                                                         imgaug.augmenters.Affine(rotate=45)]))
model.train(dataset_train, dataset_val,
            learning_rate=config.LEARNING_RATE / 200,
            epochs=300,
            layers='all',
            augmentation=augmentation)
```

**Comment by yustiks** [comment]
> You can create augmentation object and pass it to the "augmentation" field of model.train
> 
> ```
> augmentation = imgaug.augmenters.Sometimes(0.5, imgaug.augmenters.OneOf([imgaug.augmenters.Fliplr(0.5),
>                                                                          imgaug.augmenters.Flipud(0.5),
>                                                                          imgaug.augmenters.Affine(rotate=45)]))
> model.train(dataset_train, dataset_val,
>             learning_rate=config.LEARNING_RATE / 200,
>             epochs=300,
>             layers='all',
>             augmentation=augmentation)
> ```

perfect, many thanks!

**Comment by kimile599** [comment]
> > You can create augmentation object and pass it to the "augmentation" field of model.train
> > 
> > ```
> > augmentation = imgaug.augmenters.Sometimes(0.5, imgaug.augmenters.OneOf([imgaug.augmenters.Fliplr(0.5),
> >                                                                          imgaug.augmenters.Flipud(0.5),
> >                                                                          imgaug.augmenters.Affine(rotate=45)]))
> > model.train(dataset_train, dataset_val,
> >             learning_rate=config.LEARNING_RATE / 200,
> >             epochs=300,
> >             layers='all',
> >             augmentation=augmentation)
> > ```
> 
> perfect, many thanks!

have you tried that, does it work? Thr aug will aug both the pics and the label right?

**Comment by yustiks** [comment]
> > > You can create augmentation object and pass it to the "augmentation" field of model.train
> > > ```
> > > augmentation = imgaug.augmenters.Sometimes(0.5, imgaug.augmenters.OneOf([imgaug.augmenters.Fliplr(0.5),
> > >                                                                          imgaug.augmenters.Flipud(0.5),
> > >                                                                          imgaug.augmenters.Affine(rotate=45)]))
> > > model.train(dataset_train, dataset_val,
> > >             learning_rate=config.LEARNING_RATE / 200,
> > >             epochs=300,
> > >             layers='all',
> > >             augmentation=augmentation)
> > > ```
> > 
> > 
> > perfect, many thanks!
> 
> have you tried that, does it work? Thr aug will aug both the pics and the label right?

yes, that is correct

**Comment by ashwin-999** [comment]
> MRCNN supports rotation (affine) and horizontal and vertical reflections (Fliplr, Flipud)

Hello, what do you mean when you say this? Does this mean if I have complex augmentations like rotation+translation+sheer (or any other sequence of augmentations from the imgaug library), it won't work correctly?

**Comment by denashamss** [comment]
Hi everyone,
I have applied data augmentation but I dont know how can I print 
Number of dataset images  after training the model using augmentation?
